### DISCUSSION TITLE
Button Link + Icon chevron right

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1150] [Proposals]
### Goal

Possibilidade de diminuir o tamanho do Icon quando for aolicado no Button Link.

### Context

De forma cross, a Work apresentou uma proposta com o Button Link com Icon chevron right em diversos fluxos dentro no Novo App.

### Impact

Proposta Work & Co - Novo App.

### Discussion Questions

Seria possível aplicar essa condição?

### Do you have a visual proposal?

https://www.figma.com/file/seiNjy7YmPWEY3BfY4WYpp/Novos-componentes---Sprint-1?node-id=19%3A21448&t=odJ2DBpIItDwEY1S-1

**Comment by yceballost** [comment]
Según veo en el Figma, lo habéis metido como un cáracter, parte del label del botón

https://is.gd/5o9b9P

Me pregunto si esto sería una buena práctica, me aventuro a decir que no lo es puesto que en lector de pantallas se llegaría a leer ese caracter?

**Reply by BrunoBiagioniNeto** [reply]
El lector de pantalla leerá el elemento aria-label del código. 
En este caso, podemos escribir cualquier cosa en el texto siempre que la etiqueta arial-label esté en el código sin el carácter especial.
![image](https://user-images.githubusercontent.com/123083886/228314766-9594dd35-2119-46fd-bbc2-31e8ccaa82fc.png)

**Reply by annefranca** [reply]
@yceballost lembrando que a ideia seria manter o ícone, mas com um tamanho menor para chegar mais perto do conceito que a Work & Co apresentou como proposta.

**Reply by BrunoBiagioniNeto** [reply]
Exato Yaio. Para que não tenhamos que inserir isso manualmente e seguirmos os padrões globais.

**Reply by yceballost** [reply]
Hablando con el equipo, creemos hacer uso de un aria-label en el caso del botón no es una buena práctica, más cuando lo que se quiere hacer evitar hacer uso de una feature del botón que ya cubre la necesidad del chevron right (trailing icon).

Esta necesidad debería cubrirse con el trailing icon del button. 

Entendemos que el problema de esto es que el icono de chevron-right no se ajuste en tamaño y espacio a lo que buscáis visualmente.

Para esto hay varias cosas que se podría hacer para llegar al resultado deseado.
1. El trailing icon del botón admite un tamaño custom en desarrollo. Podría definirse un tamaño más pequeño para vuestro caso.
2. Añadir un icono custom para este uso del botón

Esto es un ejemplo con un icono de internet
https://rebrand.ly/aw3u0k4

Otra alternativa sería incluir en mistica-icons el icono, pero también cambiaría en las listas ya que deberíamos de tener un único tipo de chevron-right en la librería de iconos

**Reply by annefranca** [reply]
@BrunoBiagioniNeto qual das opções você acha mais relevante seguir?

**Reply by BrunoBiagioniNeto** [reply]
Vamos validar do nosso lado essa opção @yceballost

**Reply by yceballost** [reply]
habría que validar con CMS cuál es la mejor opción, Mística es flexible en esta parte, dependerá de hasta donde pueda configurarse el CMS

**Reply by yceballost** [reply]
Tengo otra duda, esto aplicaría a TODOS los buttonLink para vivo? o solo cuando se quiera?

**Reply by annefranca** [reply]
@yceballost para todos Button Link da Vivo. 

Seguem alguns links da Lib Intermediária:

https://www.figma.com/file/JkA7OefLqga6XyfiqLq1dY/Vivo---Lib-Intermedi%C3%A1ria?node-id=6%3A7857&t=Wh4vuxvDqdSjf6Vh-1

https://www.figma.com/file/JkA7OefLqga6XyfiqLq1dY/Vivo---Lib-Intermedi%C3%A1ria?node-id=1524%3A104696&t=Wh4vuxvDqdSjf6Vh-1

https://www.figma.com/file/JkA7OefLqga6XyfiqLq1dY/Vivo---Lib-Intermedi%C3%A1ria?node-id=525%3A37868&t=Wh4vuxvDqdSjf6Vh-1

**Reply by yceballost** [reply]
@BrunoBiagioniNeto llegásteis a validar algo?

Me surgen más preguntas, por qué el chevron del button es diferente al chevron de una lista?

**Reply by BrunoBiagioniNeto** [reply]
Creo que son diferentes porque la Lista es un ícono y el Enlace del botón es un carácter especial. Con la definición de usar el enlace del botón con un icono, ambos serán iguales.

**Reply by yceballost** [reply]
Buenas de nuevo!

Estamos viendo diferentes formas de abordar esto. Nos parece overkill la propuesta que tenéis, es decir, aplicar chevron a TODOS los links nos parece que puede no tener sentido en algunos casos. 

Por ejemplo
![image](https://github.com/Telefonica/mistica-design/assets/6722153/3f41458b-1fb3-437b-b7a6-ad92c5888172)

En esta caso cuando se usa un buttonLink con la idea de "eliminar" como acción, creemos que no debería existir un chevron.

Creemos que podriamos incluir una lógica dónde solo se active el chevron en aquellos casos que el ButtonLink navegue.

en términos de props sería algo asi:
`onPress` sin chevron
`submit` sin chevron
`href` con chevron
`to` con chevron

Estamos viendo con desarrollo si esto les encaja, pero os lo adelantamos para ver si a vosotros tb os encajaría

**Reply by pihcardozo** [reply]
Olá @yceballost

Discutimos as regras de uso do icon-chevron no Button Link e chegamos a conclusão:


**1** - Teremos a regra da aplicação icon-chevron no Button Link quando for uma experiência de navegação de uma página para outra.

**href com chevron
to com chevron**

Exemplos:

> Exibir mais sobre a assinatura
> Exibir todos os apps
> Exibir faturas
> Exibir detalhes
> Conhecer mais
> Entender como funciona


**2** - Ações que permanecem na mesma página não precisam ter o icon-chevron no Button Link.

**onPress sem chevron
submit sem chevron**

Exemplos:

> Selecionar todas
> Limpar selecionadas
> Cancelar
> Fechar
> Enviar



**3** - A regra se estende para o componente Button Fixed Footer Layout, onde não vamos aplicar o icon-chevron no Button Link, independente se continuar na mesma página ou ir para outra página. Exemplo:

![Captura de Tela 2023-06-05 às 16 56 47](https://github.com/Telefonica/mistica-design/assets/124393542/82b0b235-b2dc-4945-89ea-c2047650847a)

**Reply by yceballost** [reply]
Perfect, we will include a new issue to work on this definition. Thanks Fábio!
https://github.com/Telefonica/mistica-design/issues/1240
